prediction_market_agent/agents/safe_guard_agent/safe_utils.py

```diff
@@ -103,27 +103,28 @@ def sign_or_execute(safe: Safe, tx: SafeTx, api_keys: APIKeys) -> None:
     else:
         tx.sign(api_keys.bet_from_private_key.get_secret_value())
 
-    if safe.retrieve_threshold() > len(tx.signatures):
+    threshold = safe.retrieve_threshold()
+
+    if threshold > len(tx.signers):
         logger.info("Threshold not met yet, just adding a sign.")
         _abort_on_safe_owner_post(safe, tx, api_keys)
         api = TransactionServiceApi(EthereumNetwork(RPCConfig().chain_id))
         api.post_signatures(tx.safe_tx_hash, tx.signatures)
+        return None
     else:
-        logger.info("Threshold met, executing.")
+        logger.info(
+            f"Threshold {threshold} met with {len(tx.signers)} signs, executing."
+        )
         tx.call()
-        tx.execute(api_keys.bet_from_private_key.get_secret_value())
+        _, tx_params = tx.execute(api_keys.bet_from_private_key.get_secret_value())
+        return tx_params
```

Comment:
Why is the agent executing the transaction here? I would have expected the proposer to execute the transaction, the agent should be more like an "observer", no?

Reply:
It was executing it before as well. Imagine you submit a transaction in Safe App, in the case the threshold is met, as a user, would you prefer to go back and execute it manually? So every few minutes, go back and check if you can execute it already. Or for the agent to execute it for you, if the threshold is met? I like the second option more.

Reply:
I would not 100% trust the agent, as is the case for some slightly more paranoid users - but anyways, if this is the flow you prefer, works for me.